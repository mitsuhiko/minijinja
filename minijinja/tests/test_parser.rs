#![cfg(feature = "unstable_machinery")]
use minijinja::machinery::parse;

#[test]
fn test_parser() {
    insta::glob!("parser-inputs/*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let filename = path.file_name().unwrap().to_str().unwrap();
        let ast = parse(&contents, filename, Default::default(), Default::default());
        insta::with_settings!({
            description => contents.trim_end(),
            omit_expression => true,
        }, {
            insta::assert_debug_snapshot!(&ast);
        });
    });
}
