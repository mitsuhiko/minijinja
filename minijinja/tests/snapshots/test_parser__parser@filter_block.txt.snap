---
source: minijinja/tests/test_parser.rs
expression: "&ast"
input_file: minijinja/tests/parser-inputs/filter_block.txt

---
Ok(
    Template {
        children: [
            FilterBlock {
                filter: Filter {
                    name: "baz",
                    expr: Some(
                        Filter {
                            name: "bar",
                            expr: Some(
                                Filter {
                                    name: "foo",
                                    expr: None,
                                    args: [],
                                } @ 1:10-1:13,
                            ),
                            args: [
                                Const {
                                    value: 1,
                                } @ 1:18-1:19,
                                Const {
                                    value: 2,
                                } @ 1:21-1:22,
                            ],
                        } @ 1:14-1:23,
                    ),
                    args: [],
                } @ 1:24-1:27,
                body: [
                    EmitRaw {
                        raw: "...",
                    } @ 1:30-1:33,
                ],
            } @ 1:3-1:45,
        ],
    } @ 0:0-1:48,
)
